### ISSUE TITLE
FISTA has no progress option?

### ISSUE THREAD

**Issue by MorganSchmitz** [CEA-COSMIC/ModOpt#53]
follow-up on #46: when passing on the `progress` kwarg to `FISTA`, it crashes.

I believe this is because FISTA inherits `object` and not `SetUp`...?
https://github.com/CEA-COSMIC/ModOpt/blob/master/modopt/opt/algorithms.py#L224

Sorry if I'm completely off!

```~/anaconda2/envs/shapepipe/lib/python3.6/site-packages/rca-2.0-py3.6.egg/rca/rca.py in _fit(self)
    427                 weight_optim = optimalg.ForwardBackward(alpha, weight_grad, coeff_prox, cost=weight_cost,
    428                                                 beta_param=weight_grad.inv_spec_rad, auto_iterate=False,
--> 429                                                 progress=self.modopt_verb)
    430                 weight_optim.iterate(max_iter=self.nb_subiter_weights)
    431                 alpha = weight_optim.x_final

~/anaconda2/envs/shapepipe/lib/python3.6/site-packages/modopt-1.3.1-py3.6.egg/modopt/opt/algorithms.py in __init__(self, x, grad, prox, cost, beta_param, lambda_param, beta_update, lambda_update, auto_iterate, metric_call_period, metrics, linear, **kwargs)
    535         self._beta_update = beta_update
    536         if isinstance(lambda_update, str) and lambda_update == 'fista':
--> 537             fista = FISTA(**kwargs)
    538             self._lambda_update = fista.update_lambda
    539             self._is_restart = fista.is_restart

TypeError: __init__() got an unexpected keyword argument 'progress'```

**Comment by zaccharieramzi** [comment]
Actually the problem is that we pass all the `kwargs` of `ForwardBackward` to `FISTA`. We should probably accept some `dummy_kwargs` in `FISTA` to account for that, like what we did for `SetUp` (see for insights: [this](https://github.com/CEA-COSMIC/ModOpt/pull/47#discussion_r278884742) and [this](https://github.com/CEA-COSMIC/ModOpt/pull/47#discussion_r278909072)).

**Comment by zaccharieramzi** [comment]
`FISTA` shouldn't inherit `SetUp` as it is not an algorithm in the sense defined in that module. It is more a helper class in charge of computing the different parameter updates that may occur in the forward-backward.

**Comment by sfarrens** [comment]
This issues appears to be resolved, closing.

**Comment by chaithyagr** [comment]
Hey @sfarrens , I dont think this is resolved. 
If I call: 
`opt = ForwardBackward(
        x=X,
        grad=grad_op,
        prox=ProximityL2(C_kin),
        cost=None,
        auto_iterate=False,
        beta_param=beta,
        progress=False
        )`

I get the following issue still:
File "/home/chaithyagr/anaconda3/envs/pysap/lib/python3.7/site-packages/modopt-1.3.1-py3.7.egg/modopt/opt/algorithms.py", line 537, in __init__
TypeError: __init__() got an unexpected keyword argument 'progress'

I will try to get a PR for this.. Please let me know if I am way off from what is expected way to use this.

**Comment by sfarrens** [comment]
@chaithyagr You are right. There was a small bug in init for the `FISTA` class. This should be resolved in #58.

**Comment by chaithyagr** [comment]
Hmm, awesome, Ill pull in that and check my code..

**Comment by sfarrens** [comment]
@chaithyagr The changes have been merged to the master and the latest version (1.3.3) is available on PyPi. Please confirm that this resolves your problem so that I can re-close this issue, thanks.

**Comment by chaithyagr** [comment]
@sfarrens , I pulled the latest from git and ran codes.. Its working fine now.. Thanks!
